### DISCUSSION TITLE
DELETE dla /sale/responsible-producers

### DISCUSSION THREAD

**Discussion by tomylee0** [allegro/allegro-api#10271] [Suggestion - verification]
W mojej ocenie DELETE dla /sale/responsible-producers będzie absolutnie potrzebny. Jedno konto sprzedającego może mieć ponad 2 tysiące producentów. W przypadku pomyłki nie będziemy w stanie wyczyścić zasobu co doprowadzi do ogromnego bałaganu.

**Comment by PrzemyslawLukanowski** [comment]
Przepraszam za późną odpowiedź, byłem przekonany, że odpisywałem - dziękujemy za sugestię i jeśli zdecydujemy się dodać taką metodę, oczywiście poinformujemy.

**Comment by tomylee0** [comment]
Dodam, że podczas testowania dodawania responsible-producers na sandboxie nie mamy możliwości uporządkowania danych po zakończeniu testów. To uniemożliwia między innymi ponowne przeprowadzenie scenariusza testowego z pustym zasobem responsible-producers. Dlatego bardzo nalegam na DELETE.

**Comment by tomylee0** [comment]
Analogicznie potrzebne również będzie DELETE dla /sale/responsible-persons. Ilość tych obiektów też może sięgać tysiąc lub więcej.

**Reply by PrzemyslawLukanowski** [reply]
Dziękuję również za tę sugestię. Będziemy informować, jeśli zdecydujemy się wdrożyć to rozwiązanie.

**Comment by tztztztz** [comment]
Tak, to byłoby bardzo potrzebne do każdego z elementów GPSR. 

Oczywiście rozumiem, że są problemy ze spójnością - ale w przypadku usunięcia elementu GPSR, jest jeszcze sprawa powiązanych z nim ofert - i tutaj powinien się po prostu ustawiać NULL.

Problem może być taki, kiedy w pewnym momencie ta informacja będzie wymagana dla oferty, wtedy pewnie system allegro będzie zgłaszać jakiś błąd i uniemożliwi usunięcie.

Tak czy inaczej pomimo tych komplikacji moim zdaniem powinna być możliwość usuwania.

**Reply by PrzemyslawLukanowski** [reply]
Dziękuję za komentarz.

**Comment by creativepl-php** [comment]
Zdecydowanie must have. Przede wszystkim ze względu na limity producentów, ale też optymalizację. Mam na produkcji 5 producentów testowych... I póki co nigdzie się nie wybierają, a szkoda.

**Comment by tomylee0** [comment]
Chciałbym zapytać czy są planowane lub może są już rozpoczęte prace nad wdrożeniem możliwości DELETE /sale/responsibleProducer i DELETE /sale/responsiblePerson. Otrzymujemy od Allegro monity o brakujących danych związanych z GPSR. Obawiamy się jednak wprowadzać te informacje, dopóki nie będziemy mieli pewności, że w razie potrzeby będziemy mogli je usunąć i wstawić na nowo.

**Reply by PrzemyslawLukanowski** [reply]
Mamy zaplanowane dodanie takiego endpointu, jednak na tę chwilę nie mogę niestety podzielić się informacją, kiedy uda się go wdrożyć.

**Comment by SmuxSmux** [comment]
Witam, zostałem skierowany do tego wątku który jak widzę został bardzo długi czas temu stworzony.

W moim przypadku problem polega na tym że w danych producenta pozycja została ''powielona'' z zupełnie obcą mi nazwą własną. 
Musiała się podgrać automatycznie z katalogu allegro przy edycji produktu.
Ja jako użytkownik mogę jedynie takie dane edytować. Nie mam opcji usunięcia powielonej pozycji.

Jestem zdania jak moi przedmówcy że możliwość usunięcia producenta jest NIEZBĘDNA i powinna być dostępna od samego początku . Według nie są to Bardzo ważne dane , które w przypadku różnych błędów czy jakiejś pomyłki powinno się mieć możliwość ich usunięcia. Czy są już może realne plany lub plan rozpoczęcia prac, tudzież rozpoczęte prace nad wdrożeniem tej naprawdę NIEZBĘDNEJ funkcji ?

**Reply by PrzemyslawLukanowski** [reply]
Przepraszam za mocno spóźnioną odpowiedź, ale musiałem ten temat skonsultować z zespołem. Mamy w planach dodanie metody DELETE, ale na tę chwilę nie jesteśmy niestety w stanie wskazać, kiedy uda się to wykonać.

**Comment by SorentOSanerO** [comment]
Witam Serdecznie, Czy wiadomo coś może kiedy opcja zostanie wprowadzona ? Brak jej jest bardzo problematyczny i naprawdę powinna być już wprowadzona .

**Reply by PrzemyslawLukanowski** [reply]
Witam, planujemy tę opcję, jednak ze względu na aktualne priorytety, nie uda się tego zrobić w najbliższym czasie. Mamy ten temat cały czas na uwadze i jak tylko uda się to zrobić, będziemy informować o wdrożeniu.
